tests/net/webrequest.cpp

```diff
@@ -0,0 +1,213 @@
+// Author:      Tobias Taschner
+// Created:     2018-10-24
+// Copyright:   (c) 2018 wxWidgets development team
+// Licence:     wxWindows licence
+/////////////////////////////////////////////////////////////////////////////
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+#include "testprec.h"
+
+#ifdef __BORLANDC__
+#pragma hdrstop
+#endif
+
+#ifndef WX_PRECOMP
+    #include "wx/wx.h"
+#endif // WX_PRECOMP
+
+#include "wx/webrequest.h"
+#include "wx/filename.h"
+#include "wx/wfstream.h"
+
+// This test requires the URL to an httpbin instance.
+// httpbin is a HTTP Request & Response Service available at:
+// https://httpbin.org
+// It can also be run locally via a simple docker run
+//
+// For this test to run the the base URL has to be specified with
+// an environment variable, e.g.:
+//   WX_TEST_WEBREQUEST_URL=https://httpbin.org
+
+#if wxUSE_WEBREQUEST
+
+class RequestFixture
+{
+public:
+    RequestFixture()
+    {
+        expectedFileSize = 0;
+        dataSize = 0;
+    }
+
+    void Create(const wxString& subURL)
+    {
+        wxString baseURL;
+        wxGetEnv("WX_TEST_WEBREQUEST_URL", &baseURL);
+        CreateAbs(baseURL + subURL);
+    }
+
+    void CreateAbs(const wxString& url)
+    {
+        request.reset(wxWebSession::GetDefault().CreateRequest(url));
+        request->Bind(wxEVT_WEBREQUEST_STATE, &RequestFixture::OnRequestState, this);
+        request->Bind(wxEVT_WEBREQUEST_DATA, &RequestFixture::OnData, this);
+    }
+
+    void OnRequestState(wxWebRequestEvent& evt)
+    {
+        switch (evt.GetState())
+        {
+        case wxWebRequest::State_Unauthorized:
+        case wxWebRequest::State_Completed:
+            if ( request->GetStorage() == wxWebRequest::Storage_File )
+            {
+                wxFileName fn(evt.GetResponseFileName());
+                REQUIRE( fn.GetSize() == expectedFileSize );
+            }
+            wxFALLTHROUGH;
+        case wxWebRequest::State_Failed:
+        case wxWebRequest::State_Cancelled:
+            loop.Exit();
+            break;
+        }
+    }
+
+    void OnData(wxWebRequestEvent& evt)
+    {
+        // Count all bytes recieved via data event for Storage_None
+        dataSize += evt.GetDataSize();
+    }
+
+    void Run(wxWebRequest::State requiredState = wxWebRequest::State_Completed,
+        int requiredStatus = 200)
+    {
+        REQUIRE( request->GetState() == wxWebRequest::State_Idle );
+        request->Start();
+        loop.Run();
+        REQUIRE( request->GetState() == requiredState );
+        if (requiredStatus)
+            REQUIRE( request->GetResponse()->GetStatus() == requiredStatus );
+    }
+
+    wxEventLoop loop;
+    wxObjectDataPtr<wxWebRequest> request;
+    wxInt64 expectedFileSize;
+    wxInt64 dataSize;
+};
+
+TEST_CASE_METHOD(RequestFixture, "WebRequest", "[net][.]")
```

Comment:
`[.]` should be removed, as it doesn't run without the env var anyhow, it's fine to run it by default, I think.